### ISSUE TITLE
ERROR: (1208): Unable to connect to enrollment service at '[10.104.0.47]:1515'

### ISSUE THREAD

**Issue by Ekene95** [wazuh/wazuh#17519]
Hi @vikman90,

I get the error below when trying to enroll an agent.

./agent-auth -d -m 10.104.0.47 -p 1515

2023/06/10 21:58:38 agent-auth[29651] debug_op.c:70 at _log(): DEBUG: Logging module auto-initialized
2023/06/10 21:58:38 agent-auth[29651] main-client.c:248 at main(): DEBUG: Wazuh home directory: /var/ossec
2023/06/10 21:58:38 agent-auth[29651] main-client.c:291 at main(): INFO: Started (pid: 29651).
2023/06/10 21:58:38 agent-auth[29651] enrollment_op.c:122 at w_enrollment_request_key(): INFO: Requesting a key from server: 10.104.0.47
2023/06/10 21:58:38 agent-auth[29651] ssl.c:83 at os_ssl_keys(): DEBUG: Returning CTX for client.
2023/06/10 21:58:38 agent-auth[29651] enrollment_op.c:222 at w_enrollment_connect(): ERROR: (1208): Unable to connect to enrollment service at '[10.104.0.47]:1515'

Details of my setup:
One master node on Ubuntu 18.04
One worker node on Ubuntu 18.04
1 agent running on Oracle Linux 8
1 agent running on Ubuntu 18.04

I have disabled "auth" in "ossec.conf" in my worker node.

I have also followed the suggestions and several other suggestions found here and else but no luck. I even changed the enrollment port to 9999 from 1515 in all ossec.conf file but no luck either.
I tested connection to port 1515 on my Master node using #nc -zv 1515 and I got a successful output.

I am stuck now. Kindly assist. Thanks

**Comment by matias-braida** [comment]
Hello Ekene95,

First of all, thanks for using Wazuh.
Let me help you with this issue.

I will need some information to assist you:
* IP 10.104.0.47 is the address of the master wazuh-manager node?

* I need some information about your network. The agents are located on the same network as the master wazuh-manager node? 

* Could you please send me wazuh-managers service status logs?
  Run this command on each manager to obtain it and send the command output: 
  systemctl status wazuh-manager.service

* Could you please send me wazuh-managers service error logs?
  Run this command on each manager to obtain it and send the command output: 
  journalctl -xeu wazuh-manager.service

* If the manager status is ok, please obtain open ports on wazuh-manager node. Running this command and send the command output:
  sudo netstat -ltnp

* Could you please test the connection from both agents to the master wazuh-manager node running the following command? 
  Run this command on each agent having the manager running and send the command output: 
  nc -zv <MANAGER_IP> 1514 1515

Regards

**Comment by Ekene95** [comment]
Hello Matias,

Thanks for assisting me on this one.
Please see my answers inline below.

I need some information about your network.
IP 10.104.0.47 is the address of the master wazuh-manager node?
**[Ekene] That's correct. The master node is 10.104.0.47 and worker node is 10.104.0.174**
The agents are located on the same network as the master wazuh-manager node?
**[Ekene] Yes, the Agents are on the same network as the master and worker nodes. The subnet is 10.104.0.0/0.**


Could you please send me wazuh-managers service status logs?
Run this command on each manager to obtain it and send the command output:
systemctl status wazuh-manager.service
**[Ekene] Please find attached in the text file.**

Could you please send me wazuh-managers service error logs?
Run this command on each manager to obtain it and send the command output:
journalctl -xeu wazuh-manager.service
**[Ekene] Please find attached in the text file.**


If the manager status is ok, please obtain open ports on wazuh-manager node. Running this command and send the command output:
sudo netstat -ltnp
**[Ekene] Please find attached in the text file.**

Could you please test the connection from both agents to the master wazuh-manager node running the following command?
Run this command on each agent having the manager running and send the command output:
nc -zv <MANAGER_IP> 1514 1515
**[Ekene] Please find attached in the text file.**

Please note that this setup is in Oracle cloud.

Regards,
Ekene.
[wazuhAgent-troubleshooting.txt](https://github.com/wazuh/wazuh/files/11715705/wazuhAgent-troubleshooting.txt)

**Comment by Ekene95** [comment]
I made a mistake while stating my subnet. The subnet is **10.104.0.0/24** not 10.104.0.0/0

**Comment by matias-braida** [comment]
As I can see in the logs you send me, the manager is up and listening in the correct port.
Since the nc command fails to connect from the agents to the manager but also the ping command works ok, seems to be a firewall issue on the wazuh-manager (not on the agents).

Let's do a quick test, please disable the firewall on the wazuh-manager node. Running this command:
sudo ufw disable

To check that the firewall status is inactive, run this command:
sudo ufw status

Now, with the wazuh-manager firewall disabled, run again "nc -zv <MANAGER_IP> 1514 1515" from the agents. And send me the output.

After this test, reenable the manager firewall by running the command:
sudo ufw enable

Regards

**Comment by Ekene95** [comment]
Hi,

Wazuh manager
root@wazuh-1:/home/ubuntu# ufw disable
Firewall stopped and disabled on system startup
root@wazuh-1:/home/ubuntu# ufw status
Status: inactive
root@wazuh-1:/home/ubuntu# date
Sun Jun 11 21:39:01 UTC 2023


Wazuh Agent

root@wazuh-agent-2:/home/ubuntu# nc -zv 10.104.0.47 1514
nc: connect to 10.104.0.47 port 1514 (tcp) failed: No route to host
root@wazuh-agent-2:/home/ubuntu#
root@wazuh-agent-2:/home/ubuntu#
root@wazuh-agent-2:/home/ubuntu# nc -zv 10.104.0.47 1515
nc: connect to 10.104.0.47 port 1515 (tcp) failed: No route to host
root@wazuh-agent-2:/home/ubuntu#
root@wazuh-agent-2:/home/ubuntu# date
Sun Jun 11 21:39:06 UTC 2023

Done and the issue persists.

Regards,

**Comment by matias-braida** [comment]
Hi again,

As nc command failed it confirms that this is a connection problem on these ports between the agents and the manager.
Usually, disabling the wazuh-manager node's firewall should allow connection since the agents are working on the same network as the manager.
Seems to be something more that is blocking the connection to ports 1514 and 1515. 

I found some information related to OCI network security. OCI networks define security lists that act as virtual firewall for the instances. 
By default, the security lists don't allow incoming connections except for SSH and ICMP traffic. Take a look at link: https://docs.oracle.com/en-us/iaas/Content/Network/Concepts/securitylists.htm#Default
I think it is needed to modify the security list that the network is using to allow incoming traffic on wazuh ports.
After modifying the oracle security list, then a test with the wazuh-manager's firewall disabled can be run again to check if the connection is now allowed.

I also found a tutorial on how to open a port for public access, but I think that the creation of rules for the private network should be similar.
Link: https://www.youtube.com/watch?v=cHIphTiYMpw

Please tell me if this helps.
Regards

**Comment by Ekene95** [comment]
Hi Matias,

I have those ports already opened in my Security Lists for the VCN my Wazuh servers are connected to.

I made some progress after installing firewalld on all the servers. I used "systemctl stop firewalld", to disable the firewall and now I can get a connection.

1st Agent
[root@wazuh-agent opc]# nc -zv 10.104.0.47 1514
Ncat: Version 7.70 ( https://nmap.org/ncat )
Ncat: Connected to 10.104.0.47:1514.
Ncat: 0 bytes sent, 0 bytes received in 0.01 seconds.
[root@wazuh-agent opc]#
[root@wazuh-agent opc]# nc -zv 10.104.0.47 1515
Ncat: Version 7.70 ( https://nmap.org/ncat )
Ncat: Connected to 10.104.0.47:1515.
Ncat: 0 bytes sent, 0 bytes received in 0.01 seconds.


2nd Agent
root@wazuh-agent-2:/var/ossec/bin# nc -zv 10.104.0.47 1514
Connection to 10.104.0.47 1514 port [tcp/*] succeeded!
root@wazuh-agent-2:/var/ossec/bin# nc -zv 10.104.0.47 1515
Connection to 10.104.0.47 1515 port [tcp/*] succeeded!

However, the enrollment is still not working.

On the Manager is see.
root@wazuh-1:/var/ossec/bin# ./agent_control -l

Wazuh agent_control. List of available agents:
   ID: 000, Name: wazuh-1 (server), IP: 127.0.0.1, Active/Local
   ID: 006, Name: wazuh-agent-2, IP: 10.104.0.140, Never connected
   ID: 007, Name: wazuh-agent, IP: any, Never connected

List of agentless devices:

root@wazuh-1:/var/ossec/bin#


On the Agents, the errors in the logs are the same.

2023/06/12 07:12:03 wazuh-agentd: INFO: Requesting a key from server: 10.104.0.47
2023/06/12 07:12:03 wazuh-agentd: INFO: No authentication password provided
2023/06/12 07:12:03 wazuh-agentd: INFO: Using agent name as: wazuh-agent-2
2023/06/12 07:12:03 wazuh-agentd: INFO: Waiting for server reply
2023/06/12 07:12:03 wazuh-agentd: ERROR: Duplicate agent name: wazuh-agent-2 (from manager)
2023/06/12 07:12:03 wazuh-agentd: ERROR: Unable to add agent (from manager)
2023/06/12 07:12:13 wazuh-agentd: WARNING: (4101): Waiting for server reply (not started). Tried: '10.104.0.47'.
2023/06/12 07:12:13 wazuh-agentd: WARNING: Unable to connect to any server.
2023/06/12 07:12:13 wazuh-agentd: INFO: Closing connection to server ([10.104.0.47]:1515/tcp).
2023/06/12 07:12:13 wazuh-agentd: INFO: Trying to connect to server ([10.104.0.47]:1515/tcp).
2023/06/12 07:12:23 wazuh-agentd: INFO: Closing connection to server ([10.104.0.47]:1515/tcp).
2023/06/12 07:12:23 wazuh-agentd: INFO: Trying to connect to server ([10.104.0.47]:1515/tcp).
2023/06/12 07:12:33 wazuh-agentd: INFO: Closing connection to server ([10.104.0.47]:1515/tcp).
2023/06/12 07:12:33 wazuh-agentd: INFO: Trying to connect to server ([10.104.0.47]:1515/tcp).

Do you know what the issue could be?

Thanks & Regards,
Ekene.

**Comment by matias-braida** [comment]
Hello Ekene,

Glad to hear that now communication is working.

Error "Duplicate agent name: wazuh-agent-2 (from manager)" means that the manager already has a registered agent with the name "wazuh-agent-2" so it can't register a new one with the same name.

* First action: delete the existing registered agent on the manager.

You could run this command in your Wazuh Manager to list all your registered agents:
/var/ossec/bin/manage_agents -l 

And then delete the one you need using its ID:
/var/ossec/bin/manage_agents -r 006

Documentation link: https://documentation.wazuh.com/current/user-manual/reference/tools/manage-agents.html

* Second action: enroll again the agent.

You can enroll the agent manually or automatically.

Manual enrollment:

Run this command on the agent you want to register:
/var/ossec/bin/agent-auth -m <manager_IP> -A <agent_name>

Please ensure that the "agent_name" is not already registered in the manager.
Documentation link: https://documentation.wazuh.com/current/user-manual/reference/tools/agent-auth.html

Automatic mode:

If you use this method, please check that the agent configuration file ossec.conf does not have an already registered agent name in the section client->enrollment->agent_name.
Please take a look at the link: https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/client.html#agent-name

Please tell me if this helps.
Regards

**Comment by Ekene95** [comment]
Hello Matias,

Thanks for your response.

I have removed the agents but it keeps re-adding automatically. For now, I stopped the agent service on the agent servers using " systemctl stop agent.service", to prevent auto-enrolling..

I have cleared the files in /var/ossec/queue/rids/ directory, from a troubleshooting step I saw online.

Also, I don't have the <enrollment> tag in my agent "ossec.conf" file. Here is an excerpt from one of the agents.

<ossec_config>
  <client>
    <server>
      <address>10.104.0.47</address>
      <port>1515</port>
      <protocol>tcp</protocol>
    </server>
    <config-profile>ubuntu, ubuntu18, ubuntu18.04</config-profile>
    <notify_time>10</notify_time>
    <time-reconnect>60</time-reconnect>
    <auto_restart>yes</auto_restart>
    <crypto_method>aes</crypto_method>
  </client>

Please let me know if you have further suggestions. Thanks.

Regards,
Ekene.

**Comment by Ekene95** [comment]
Hi Matias,

I eventually deleted the VMs for the 2 agents and instantiated another one.

On the new VM, I installed and started the agent and it was successful this time. I attached a screenshot.

I would have sent you the logs but I deleted the VM and its boot volume as well.

I still don't understand why the agent kept re-enrolling anytime I remove it from the Manager. It takes only a few seconds to get re-enrolled and the enrolling process never completes, leaving the Agent in "Never Connected" state.

Thanks a lot for your time and assistance. Much appreciated!
![wazuh-agent](https://github.com/wazuh/wazuh/assets/61330257/10e8829f-1d15-4c82-809b-dbbc276652e4)

Regards,
Ekene.
